Refactor template code to increase readability

kreha1 opened the pull request:
This is a minor refactor done in order to ease the transition to windicss. I wish to rewrite the whole project with typescript, fully transition to the composition api using script setup and convert mixins to composable functions.

I'd also like to move from webpack to [vite](https://vitejs.dev/guide/why.html#slow-server-start), as this is where all the nice things are at this moment (like pretty much instant dev server start, and plenty of plugins), as well as to move from yarn (slower, FB owned) to [pnpm](https://pnpm.io/).

I wanted to limit changes to mostly HTML, but there are many places where logic itself could be improved by using optional chaining, nullish coalescing operator, or simply using await keyword instead of assigning values inside `then` callback.

If you are okay with my contribution, let me know in the comment.

kreha1 commented on src/components/Playlist.vue:
```diff
@@ -2,24 +2,24 @@
-                        {{ playlist.uploader }}</router-link
-                    ></b
-                >
+                        <span v-text="playlist.uploader" />
```
```suggestion
                        <strong v-text="playlist.uploader" />
```

kreha1 commented on src/components/Playlist.vue:
```diff
@@ -2,24 +2,24 @@
             <div>
-                <b
-                    ><router-link class="uk-text-justify" :to="playlist.uploaderUrl || '/'">
+                <strong>
```
```suggestion
```

kreha1 commented on src/components/Playlist.vue:
```diff
@@ -2,24 +2,24 @@
-                >
+                        <span v-text="playlist.uploader" />
+                    </router-link>
+                </strong>
```
```suggestion
```

kreha1 reviewed (commented):
I've tracked some leftovers.

FireMasterK commented:
Thank you for the pull request!

> I wish to rewrite the whole project with typescript, fully transition to the composition api using script setup and convert mixins to composable functions.

I'm not familiar with typescript, but I'm willing to learn :)

> I'd also like to move from webpack to vite

This would be nice

> as well as to move from yarn to pnpm

What happens to dependabot then? I don't think it supports pnpm sadly

> optional chaining, nullish coalescing operator

I believe these require webpack 5 :thinking:, not sure what happens if we use vite tho

> If you are okay with my contribution, let me know in the comment.

Yes, I had a look at these changes, it looks good to me!

FireMasterK commented on package.json:
```diff
@@ -29,15 +29,16 @@
         "@vue/cli-plugin-pwa": "^4.5.15",
         "@vue/cli-service": "^4.5.15",
         "@vue/compiler-sfc": "3.2.26",
+        "@vue/eslint-config-prettier": "^6.0.0",
         "babel-eslint": "^10.1.0",
-        "eslint": "^7.32.0",
-        "eslint-config-prettier": "^8.3.0",
-        "eslint-plugin-prettier": "^4.0.0",
-        "eslint-plugin-vue": "^7.20.0",
+        "eslint": "^6.7.2",
+        "eslint-plugin-prettier": "^3.3.1",
+        "eslint-plugin-vue": "^7.0.0",
+        "prettier": "^2.2.1",
```
Why are we downgrading some dependencies?
eg: eslint `7.32.0` -> `6.7.2`

kreha1 replied:
npm and pnpm couldn't resolve dependencies when I first tried fetching packages, due to `@vue/cli-plugin-eslint@4.5.15` having set `eslint@">= 1.6.0 < 7.0.0"` as a peer dependency. Yarn doesn't actually care, so I can as well revert this commit. Especially since when will have switched to vite, this dependency will no longer be there.

Would you prefer a commit reverting the changes, or to leave it out during a rebase? The latter will require a force push, but that won't have any implications beyond this PR.

FireMasterK replied:
A revert works just as fine :)

kreha1 commented:
> > as well as to move from yarn to pnpm
>
> What happens to dependabot then? I don't think it supports pnpm sadly

So this will be on-hold until they do support it. This should be a matter of time, as [Renovate](https://github.com/renovatebot/renovate/) already supports pnpm.

> > optional chaining, nullish coalescing operator
> 
> I believe these require webpack 5 🤔, not sure what happens if we use vite tho

Vite uses ES modules by default, so it's only a matter of configuring babel if it's even needed as [Optional Chaining](https://caniuse.com/mdn-javascript_operators_optional_chaining) and [Nullish Coalescing Operator](https://caniuse.com/mdn-javascript_operators_nullish_coalescing) have been supported by all major browser for more than a year already.

FireMasterK commented:
I've just migrated from Webpack to Vite in https://github.com/TeamPiped/Piped/pull/683/commits/562f139ad548c5375846fd8afeac2934ba0b8ca2, however, some work is needed to get PWA support and eslint working.

kreha1 commented:
I have rebased my branch onto your vite commit, so there are no conflicts (I've also dropped the first commit which changed eslint version). I've additionally made it so it passes the github action check, as when replacing vue-cli with vite, `lint` script in package.json had to be added manually.

If there are no issues on the part of my code, please merge it, so I can focus on other improvements ;).
